lib/netsuite/rest/utilities/roles.rb

```diff
@@ -0,0 +1,79 @@
+require 'faraday'
+require 'json'
+
+module NetSuite
+  module Rest
+    module Utilities
+      class Roles
+
+        API = "https://rest.netsuite.com"
+
+        def self.get(credentials)
+          new(credentials).get
+        end
+
+        def initialize(credentials)
+          @email     = encode_email(credentials.fetch(:email))
+          @signature = credentials.fetch(:password)
+        end
+
+        def get
+          response = connection.get("/rest/roles")
+          parsed   = JSON.parse(response.body)
+          return parsed unless response.success?
+          extract(parsed)
+        end
+
+        private
+
+        def encode_email(unencoded_email)
+          URI.escape(
+            unencoded_email,
+            Regexp.new("[^#{URI::PATTERN::UNRESERVED}]")
+          )
+        end
+
+        def connection
+          Faraday.new(API) do |faraday|
+            faraday.request        :url_encoded
+            faraday.adapter        Faraday.default_adapter
+            faraday.authorization  :NLAuth,
+              nlauth_email:        @email,
+              nlauth_signature:    @signature
+          end
+        end
```

Comment:
@davidlaprade I love the idea of building this into the gem, but I'm hesitant to include a external gem dependency for a single HTTP request. Extra dependencies == extra headaches for developers trying to upgrade other gems.

Given the simplicity of this request, I'd prefer to use a ruby stdlib http client like `Net::HTTP`.

What do you think?